kafka/src/main/java/org/apache/apex/malhar/kafka/AbstractKafkaPartitioner.java

```diff
@@ -257,13 +257,13 @@ public boolean equals(Object o)
       }
       PartitionMeta that = (PartitionMeta)o;
       return Objects.equals(cluster, that.cluster) &&
-        Objects.equals(topicPartition, that.topicPartition);
+        Objects.equals(getTopicPartition(), that.getTopicPartition());
```

Comment:
instead of comparing topicPartition object, why not compare topic and partitionId?